core/src/main/java/com/datastax/oss/driver/internal/core/metadata/token/TokenRangeBase.java

```diff
@@ -275,7 +274,7 @@ public boolean equals(Object other) {
 
   @Override
   public int hashCode() {
-    return Objects.hash(start, end);
+    return 31 * start.hashCode() + end.hashCode();
```

Comment:
According to [List.hashCode()](https://docs.oracle.com/javase/8/docs/api/java/util/List.html#hashCode--)

```
     int hashCode = 1;
     for (E e : list)
         hashCode = 31*hashCode + (e==null ? 0 : e.hashCode());
```
It seems to me that it should be ` (31 + start.hashCode()) * 31 + end.hashCode()`.